code/datums/elements/footstep.dm

```diff
@@ -74,31 +74,41 @@
 			playsound(turf, 'sound/effects/footstep/crawl1.ogg', 15 * volume, falloff_distance = 1, vary = sound_vary)
 		return
 
-	if(iscarbon(source))
-		var/mob/living/carbon/carbon_source = source
-		if(!carbon_source.get_bodypart(BODY_ZONE_L_LEG) && !carbon_source.get_bodypart(BODY_ZONE_R_LEG))
-			return
-		if(carbon_source.move_intent == MOVE_INTENT_WALK)
-			return// stealth
+	if(iscarbon(source) && source.move_intent == MOVE_INTENT_WALK)
+		return // stealth
+
 	steps_for_living[source] += 1
 	var/steps = steps_for_living[source]
 
-	if(steps >= 6)
+	if(steps >= 8)
```

Comment:
Considering that while floating, the sound is only played every six steps, I think this should be 24 (common multiple of both) to allow both left foot and right foot sounds to play evenly.

```suggestion
	if(steps >= 24)
```
Feel free to update the comment below however you want.